homeassistant/components/swiss_public_transport/sensor.py

```diff
@@ -112,37 +113,49 @@ class SwissPublicTransportSensor(SensorEntity):
     _attr_attribution = "Data provided by transport.opendata.ch"
     _attr_icon = "mdi:bus"
 
-    def __init__(self, opendata, start, destination, name):
+    def __init__(self, opendata, start, destination, name) -> None:
         """Initialize the sensor."""
         self._opendata = opendata
         self._name = name
         self._from = start
         self._to = destination
-        self._remaining_time = None
+        self._remaining_time: timedelta | None = None
+
+        self._connected = False
 
     @property
-    def name(self):
+    def name(self) -> str:
         """Return the name of the sensor."""
         return self._name
 
     @property
-    def native_value(self):
+    def available(self) -> bool:
+        """Return the available of the sensor."""
+        return self._connected
+
+    @property
+    def native_value(self) -> str | None:
         """Return the state of the sensor."""
         return (
             self._opendata.connections[0]["departure"]
-            if self._opendata is not None
+            if self._opendata is not None and len(self._opendata.connections) > 0
```

Comment:
Let me first ask this:

When can `self_opendata` be None?

Reply:
I think you are right, I have assumed it should be there as it was already present in the codebase when I started working on it, but I think as well that it should not be possible to have it `undefined`.

Reply:
I know you already have 1 other PR for this integration. Mind making one first where we get the code quality up to par and make use of some nice new code tricks to make it nice?

Reply:
Yes, my plan was to first improve the existing integration before adding new parameters and options. With this PR I wanted to remove an error I have seen in the logs which happened sometimes, when the sensor was asked data before he was able to fetch it. Hence the `availability` property implementation.

Reply:
What else would you need for the code quality? I am just looking at the medals, what they define, but I would be keen to know if there is a more in-depth guide to it :)

Reply:
There really isn't any guide for it. But feel free to reach out to the HA discord if you need help with stuff :)

Reply:
Ok, thank you very much!